File: examples/conn-test.js
#!/usr/bin/env gjs

const lv = imports.gi.LibvirtGObject;
const gio = imports.gi.Gio;
const gtk = imports.gi.Gtk;

lv.init_object(null, null);

var conn = new lv.Connection({ uri: "test:///default" })
var canc = new gio.Cancellable()

function done(conn, result, data) {
    try {
	conn.open_finish(result)

	print("Opened " + conn.get_uri())

	conn.fetch_domains(null)
	print ("Fetched")
	var doms = conn.get_domains()
	print ("Got " + doms)

	for (var d in doms) {
	    print ("One dom: " + doms[d])
	    print ("Name " + doms[d].get_name())
	    var conf = doms[d].get_config(0)
	    print ("Conf " + conf)
	    var xml = conf.to_xml()
	    print ("XML " + xml)
	    print ("Info " + doms[d].get_info().memory)
	}

    } finally {
	gtk.main_quit()
    }
}

conn.open_async(canc, done, null)

gtk.main(null, null)

File: examples/conn-test.py
#!/usr/bin/env python3

import gi

gi.require_version('LibvirtGObject', '1.0')
from gi.repository import LibvirtGObject
from gi.repository import Gio
gi.require_version('Gtk', '3.0')
from gi.repository import Gtk

LibvirtGObject.init_object_check(None)

conn = LibvirtGObject.Connection(uri="test:///default")
canc = Gio.Cancellable()

def done(conn, result, data):
    try:
        conn.open_finish(result)

        print("Opened " + conn.get_uri())

        conn.fetch_domains(None)
        print ("Fetched")
        doms = conn.get_domains()
        print ("Got " + str(doms))

        for d in doms:
            print ("One dom: " + str(d))
            print ("Name " + d.get_name())
            conf = d.get_config(0)
            print ("Conf " + str(conf))

            try:
                conf.validate()

                print ("Document is valid according to %s" % conf.get_schema())
            except Exception as e:
                print ("Document is not valid according to %s: %s: %s" % (conf.get_schema(), str(e), str(type(e))))

            xml = conf.to_xml()
            print ("XML " + xml)
            print ("Info " + str(d.get_info().memory))

    finally:
        Gtk.main_quit()


conn.open_async(canc, done, None)

Gtk.main()
